wiring/src/spark_wiring_ble.cpp

```diff
@@ -183,47 +183,41 @@ hal_ble_addr_t BleAddress::halAddress() const {
 }
 
 String BleAddress::toString(bool stripped) const {
-    String string;
-    uint8_t temp[BLE_SIG_ADDR_LEN];
-    toBigEndian(temp);
-    char* cStr = (char*)malloc(BLE_SIG_ADDR_LEN * 2 + 6);
-    if (cStr) {
-        SCOPE_GUARD ({
-            free(cStr);
-        });
-        if (stripped) {
-            bytes2hexbuf(temp, BLE_SIG_ADDR_LEN, cStr);
-            cStr[BLE_SIG_ADDR_LEN * 2] = '\0';
-        } else {
-            uint8_t idx = 0;
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx]);
-            idx++;
-            cStr[idx * 2] = ':';
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx * 2 + 1]);
-            idx++;
-            cStr[idx * 2 + 1] = ':';
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx * 2 + 2]);
-            idx++;
-            cStr[idx * 2 + 2] = ':';
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx * 2 + 3]);
-            idx++;
-            cStr[idx * 2 + 3] = ':';
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx * 2 + 4]);
-            idx++;
-            cStr[idx * 2 + 4] = ':';
-            bytes2hexbuf(&temp[idx], 1, &cStr[idx * 2 + 5]);
-            idx++;
-            cStr[idx * 2 + 5] = '\0';
-        }
-        string.concat(cStr);
-    }
-    return string;
+    char cStr[BLE_SIG_ADDR_LEN * 2 + 6];
+    size_t len = toString(cStr, sizeof(cStr), stripped);
+    return String(cStr, len);
 }
 
 size_t BleAddress::toString(char* buf, size_t len, bool stripped) const {
-    String string = toString(stripped);
-    len = std::min(len, string.length());
-    memcpy(buf, string.c_str(), len);
+    if (!buf || len == 0) {
+        return 0;
+    }
+    uint8_t temp[BLE_SIG_ADDR_LEN];
+    char cStr[BLE_SIG_ADDR_LEN * 2 + 5];
```

Comment:
Why not directly use passed `buf`?

Reply:
I was going to. But image that the length of given buffer is an odd number, let's say 5 bytes, then what is the length argument of the `bytes2hexbuf(temp, BLE_SIG_ADDR_LEN, cStr);` supposed to be? And it may corrupt the RAM.